tests/subsys/pm/power_states_api/src/main.c

```diff
@@ -63,5 +78,31 @@ ZTEST(power_states_1cpu, test_power_states)
 		     "Invalid pm_state_info array");
 }
 
+ZTEST(power_states_1cpu, test_device_power_state_constraints)
+{
+	static const struct device *const dev =
+		DEVICE_DT_GET(DT_NODELABEL(test_dev));
+
+	ongoing_operation = true;
+
+	test_driver_async_operation(dev);
+
+	/** Lets sleep long enough to suspend the CPU with `suspend to ram`
+	 *  power state. If everything works well the cpu should not use this
+	 *  state due the constraint set by `test_dev`.
+	 */
+	k_sleep(K_USEC(60000));
+
```

Comment:
I think `zaasert(suspend_to_ram_count == 0);` should be here. Or am I missing something?

Reply:
not really needed. The ongong_operation ensures that if suspend to ram is used an assert will happen in pm_state_set. But it is clear having it here, I will add it.